update/Pos/Update/Poll/Logic/Base.hs

```diff
@@ -200,31 +200,29 @@ updateSlottingData
     :: (MonadError PollVerFailure m, MonadPoll m)
     => EpochIndex -> m ()
 updateSlottingData epoch = do
-    sd@SlottingData {..} <- getSlottingData
     let errFmt =
             ("can't update slotting data, stored penult epoch is "%int%
              ", while given epoch is "%int%
              ")")
-    if | sdPenultEpoch + 1 == epoch -> updateSlottingDataDo sd
+    let err = throwError $ PollInternalError $ sformat errFmt (-1) epoch
+    mres <- do
+      li <- getEpochLastIndex
+      fmap (li,) <$> getEpochSlottingData li
+    (lastIndex, esd) <- maybe err return mres
```

Comment:
Also I think that the error here should be different, it should explicitly say that there is no slotting data for the epoch returned by `getEpochLastIndex`. It's different from saying that our stored penult epoch is (-1), it means that some internal invariant should be violated.
But maybe this case will disappear completely, because it makes sense to ensure that there is `EpochSlottingData` for the last index at lower-level.